TrackingTools/KalmanUpdators/src/KFUpdator.cc

```diff
@@ -7,6 +7,53 @@
 #include "DataFormats/Math/interface/invertPosDefMatrix.h"
 #include "DataFormats/Math/interface/ProjectMatrix.h"
 
+
+// test of joseph form
+#ifdef KU_JF_TEST
+
+#include<atomic>
+#include<iostream>
+namespace {
+  struct Stat {
+    Stat(): tot(0),
+    nopd(0), jnopd(0), fnopd(0),
+    inopd(0), ijnopd(0), ifnopd(0),dtot(0)
+    {for (int i=0;i<5;++i) idmm[i]=0; }
+
+    std::atomic<long long> tot;
+    std::atomic<long long> nopd;
+    std::atomic<long long> jnopd;
+    std::atomic<long long> fnopd;
+    std::atomic<long long> inopd;
+    std::atomic<long long> ijnopd;
+    std::atomic<long long> ifnopd;
+
+    std::atomic<unsigned long long> dtot;
+    std::atomic<unsigned long long> idmm[5];
+    double mm = 0;
+    double dmm[5] = {0};
+    bool ok=true;
+    ~Stat() {
+       double n = 1.e-3/double(dtot);
+       std::cout << "KF " << tot*1.e-9 << "G " << mm <<" "
```

Comment:
this IS thread safe as there is a single instance of Stat (stat)
and the destructor run when the library is "unloaded"